### PR TITLE
[FAQ] Design guidelines

### PR DESCRIPTION
**Pull request by josepmartins**
## Summary

Add `FAQ` design guidelines.

## List of notable changes:

- **added** new ComponentLayout to `FAQ` docs
- **added** design guidelines

## What should reviewers focus on?

- 👀 Documentation: https://primer-666c66d334-26139705.drafts.github.io/components/FAQ

## Supporting resources (related issues, external links, etc):

- Epic: https://github.com/github/primer/issues/2231
- Tracking issue: https://github.com/github/primer/issues/1984

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

File: apps/docs/content/components/FAQ/index.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+title: FAQ
+description: Display content in a Q&A format
+---
+
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+## Anatomy
+
+![The text component has a heading, optional subheading, and a list of questions and answers.](https://github.com/primer/brand/assets/912236/b2fa1bb7-938c-42ed-a549-fdc9f2759184)
```
Comment (jesskuo4) [inline apps/docs/content/components/FAQ/index.mdx:11]:
```suggestion
![The FAQ component has a heading, optional subheading, and a list of questions that are interactive to expand and show answers](https://github.com/primer/brand/assets/912236/b2fa1bb7-938c-42ed-a549-fdc9f2759184)
```

File: apps/docs/content/components/FAQ/index.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+- Heading: The main title of the FAQ.
+- Subheading: An optional short description that extends the information provided by the heading and helps group the items in the list.
+- Question: The title of the question.
+- Answer: The answer to the question.
+
+## Usage
+
+Use the FAQ to provide answers to common questions about a product or a feature presented on a page. The FAQ is a good way to provide additional information that is not directly related to the main content of the page. We recommend placing the FAQ at the bottom of the page, after the main content. Avoid using the FAQ to provide information that is already available in the page.
```
Comment (jesskuo4) [inline apps/docs/content/components/FAQ/index.mdx:20]:
```suggestion
Use the FAQ to provide answers to common questions about a product or a feature presented on a page. The FAQ is a good way to provide additional information that may be beyond the scope of the main content of the page. We recommend placing the FAQ at the bottom of the page, after the main content. Avoid using the FAQ to provide information that is already available in the page.
```

File: apps/docs/content/components/FAQ/index.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+Answers should be clear and be no longer than 2-3 paragraphs. If the answer is longer than that, needs visual assets or a specific layout treatment, consider using a different component. For instance, if the answer needs an image, we recommend using a list of [cards](/components/Card) or a [river](/components/River) component instead.
```
Comment (jesskuo4) [inline apps/docs/content/components/FAQ/index.mdx:22]:
```suggestion
Answers should be clear and be no longer than 2-3 paragraphs. If the answer is longer than the suggested, needs visual assets, or a specific layout treatment, consider using a different component. For instance, if the answer needs an image, we recommend using a list of [cards](/components/Card) or a [river](/components/River) component instead.
```

File: apps/docs/content/components/FAQ/index.mdx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+<DoDontContainer stacked>
+  <Do>
+    <img
+      src="https://github.com/primer/brand/assets/912236/824ebdee-4cc0-425f-8521-2c4e7b06491f"
+      alt=""
+    />
+    <Caption>
+      Use the FAQ to provide clear answers to common questions. Link to other
+      resources or use ordered/unordered lists if necessary.
```
Comment (jesskuo4) [inline apps/docs/content/components/FAQ/index.mdx:34]:
```suggestion
      resources and use ordered/unordered lists if necessary.
```

**Review by jesskuo4 — APPROVED** [review]

File: apps/docs/content/components/FAQ/index.mdx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+## Anatomy
+
+![The FAQ component has a heading, optional subheading, and a list of questions that are interactive to expand and display answers](https://github.com/primer/brand/assets/912236/b2fa1bb7-938c-42ed-a549-fdc9f2759184)
+
+- Heading: The main title of the FAQ.
+- Subheading: An optional short description that extends the information provided by the heading and helps group the items in the list.
+- Question: The title of the question.
```
[the remark this answers is not in the document]
Comment (josepmartins) [reply apps/docs/content/components/FAQ/index.mdx:15]: good point!

**Review by mperrotti — APPROVED** [review]

**Comment by mperrotti** [comment]
I wasn't able to load the preview deployment, but this looks good based on reviewing the raw `.mdx`

**Pull request merged**